include/pops/simulation.hpp

```diff
@@ -607,6 +726,11 @@ class Simulation
                         outside_dispersers.emplace_back(std::make_tuple(row, col));
                         continue;
                     }
+                    if (soil_pool_
+                        && soil_pool_->active() /* && to_soil_probabiliy */) {
+                        soil_pool_->disperser_to(i, j, generator_);
+                        continue;
+                    }
```

Comment:
These are taken from the general disperser pool by ratio, but then applied only to the same cell the disperser is in ("trivial kernel"). This can be done right away in generate for performance reasons.

Reply:
Soil is acting as an additional disperser storage besides the standard disperers pool, so storing now happens after generation in the generate function.